ttnn/cpp/ttnn/operations/conv/conv2d/prepare_conv2d_weights.cpp

```diff
@@ -252,23 +235,27 @@ std::pair<ttnn::Tensor, std::optional<ttnn::Tensor>> prepare_conv_weights_biases
         weight_tensor_ = ttnn::operations::core::to_device(weight_tensor_, device, std::nullopt);
 
     if (bias_tensor.has_value()) {
-        if (!is_non_tile_mul_width) {
-            bias_tensor_ = bias_tensor.value();
-            auto bias_shape = bias_tensor_.get_shape();
-            TT_ASSERT(bias_shape[3] == out_channels && bias_shape[0] == 1 && bias_shape[1] == 1 && bias_shape[2] == 1);
-            tt::tt_metal::LegacyShape bias_channels_padded_shape = tt::tt_metal::LegacyShape(
-                std::array<uint32_t, 4>({1, 1, 32, round_up(out_channels, weight_block_w_ntiles * 32)}));
-            bias_tensor_ = ttnn::pad(bias_tensor_, bias_channels_padded_shape.to_array_4D(), tt::tt_metal::Array4D{0, 0, 0, 0}, 0);
-            bias_tensor_ = ttnn::to_layout(
-                bias_tensor_, Layout::TILE, std::nullopt, std::nullopt, (T*)nullptr);
-            if (bias_tensor_.get_dtype() != weights_bias_dtype) {
-                bias_tensor_ = ttnn::to_dtype(bias_tensor_, weights_bias_dtype);
+        bias_tensor_ = bias_tensor.value();
+        bool is_bias_tensor_is_on_device = ttnn::is_tensor_on_device_or_multidevice(bias_tensor_);
+        if(!is_bias_tensor_is_on_device) {
+            validate_bias_tensor(bias_tensor_);
+            if (!is_non_tile_mul_width) {
+                auto bias_shape = bias_tensor_.get_shape();
+                TT_ASSERT(bias_shape[3] == out_channels && bias_shape[0] == 1 && bias_shape[1] == 1 && bias_shape[2] == 1);
```

Comment:
Are you aware that TT_ASSERT is never run in release / on CI? 
I recommend changing to TT_FATAL